JOSESwift/Sources/CryptoImplementation/EC.swift

```diff
@@ -95,33 +95,46 @@ public enum ECCurveType: String, Codable {
     }
 }
 
-fileprivate extension SignatureAlgorithm {
-    typealias DigestFunction = (
-            UnsafeRawPointer?,
-            UInt32,
-            UnsafeMutablePointer<UInt8>?
-    ) -> UnsafeMutablePointer<UInt8>?
-
-    func createDigest(input: Data) throws -> [UInt8] {
-        guard
-                let computedDigestLength = digestLength,
-                let computedDigestFunction = digestFunction
-                else {
-            throw ECError.invalidCurveDigestAlgorithm
+// converting integers to and from DER encoded ASN.1 as described here:
+// https://docs.microsoft.com/en-us/windows/desktop/seccertenroll/about-integer
+fileprivate extension Data {
+    func withLengthFixedTo(_ octetLength: Int) -> Data {
+        let varLength = self.count
+        if varLength > octetLength + 1 {
+            fatalError("Unable to parse ASN.1 Integer")
```

Comment:
This will only happen Apple violates the contract of their `SecKeyCreateSignature` function (or we made an implementation error), user code cannot trigger this condition.

How do you handle such cases elsewhere?